### PR TITLE
Add 401 to the explicitly defined codes

### LINKED ISSUE #108: Adding 401 to the explicitly defined codes
As we might reject received events in case of invalid/missing authentication and return a `401`, we should update the [spec](https://github.com/knative/specs/blob/main/specs/eventing/data-plane.md#event-acknowledgement-and-delivery-retry) to include this HTTP status code.

**References:**
* https://github.com/knative/eventing/issues/7289
* https://github.com/knative/eventing/issues/7290
* https://github.com/knative/eventing/issues/7291

### PR DESCRIPTION
**Pull request by xiangpingjiang**
# Changes

/kind <kind>

Fixes #108 

**Release Note**

```release-note

```

**Docs**

```docs
Add 401 to the explicitly defined codes
```

### CODE REVIEW

File: specs/eventing/data-plane.md [hunk 1]
```diff
@@ -108,6 +108,7 @@ treated as a retriable error.
 | other `2xx`   | (Unspecified)                                     | No    | Yes                | No    |
 | `3xx`         | (Unspecified)                                     | No\*  | No\*               | Yes\* |
 | `400`         | Unparsable event                                  | No    | No                 | Yes   |
+| `401`         | Reject received events in case of invalid / missing authentication | No    | No                 | Yes   |
```
Comment (creydr) [inline specs/eventing/data-plane.md:111]:
```suggestion
| `401`         | Invalid / missing authentication         | No    | No                 | Yes   |
```
I would keep it shorter

**Comment by ReToCode** [comment]
PTAL @knative/steering-committee, @knative/technical-oversight-committee

**Comment by creydr** [comment]
@aliok @dsimansk could you PTAL at this?

/assign @aliok @dsimansk

**Comment by creydr** [comment]
@pierDipi could you PTAL?

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Comment by pierDipi** [comment]
I have no power to approve here :)

**Comment by psschwei** [comment]
I'm assuming based on the discussion in https://github.com/knative/community/issues/1399 that @knative/technical-oversight-committee owns changes to the spec now. 

And we probably should vote on spec changes... we can probably do that async, but I'll add it to the TOC backlog too just in case.

+1 from me

**Comment by dprotaso** [comment]
+1 from me

**Comment by dsimansk** [comment]
+1

**Pull request merged**